src/ahp/verifier.rs

```diff
@@ -55,10 +55,23 @@ impl<F: PrimeField> AHPForR1CS<F> {
         let domain_k = get_best_evaluation_domain::<F>(index_info.num_non_zero)
             .ok_or(SynthesisError::PolynomialDegreeTooLarge)?;
 
-        let alpha = sample_element_outside_domain(&domain_h, rng);
-        let eta_a = F::rand(rng);
-        let eta_b = F::rand(rng);
-        let eta_c = F::rand(rng);
+        let mut alpha: F = fs_rng.squeeze_128_bits_challenge();
+
+
+        // TODO: This way of repeatedly sampling until outside the domain is 
+        //       not circuit friendly. In future we need to decide whether to
+        //       ignore it (the probability of repetition is astronomically
+        //       low) or explicitly randomize the prover (e.g. by an additional 
+        //       seed used to init the Fiat-Shamir rng) so that even a non-zk       
+        //       prover produces fresh challenges on every new run.   
+        while domain_h.evaluate_vanishing_polynomial(alpha).is_zero() {
```

Comment:
if we endup in this situation, the verifier must iterate until the value is out of the domain and this could be a problem if the verifier is in a circuit. Considering the negligible probability (@UlrichHaboeck75 ) of this to happen, let's abort proof generation instead of generating a proof that will not be verifiable in a circuit.